File: base/grafite/.cache/lang/ts/pencil.ts
import { Lead } from "./lead";

class Pencil {
    private thickness: number;
    private tip: Lead | null;

    public constructor(thickness: number) { 
    }

    public hasLead(): boolean {
        return false;
    }

    public insert(lead: Lead): boolean {
        return false;
    }

    public remove(): Lead | null {
        return null;
    }

    writePage(): void {
    }

    public toString(): string {
        let ponta = this.tip != null ? "[" + this.tip.toString() + "]" : "null";
        return "calibre: " + this.thickness + ", grafite: " + ponta;
    }

}

export { Pencil };

File: base/grafite/.cache/lang/java/Pencil.java
public class Pencil {
    private float thickness;
    private Lead tip;

    public Pencil(float thickness) {
    }

    public float getThickness() {
        return 0.0f;
    }

    public void setThickness(float value) {
    }

    public boolean hasLead() {
        return false;
    }

    public boolean insert(Lead grafite) {
        return false;
    }

    public Lead remove() {
        return null;
    }

    public void writePage() {
    }
    
    public String toString() {
        String saida = "calibre: " + thickness + ", grafite: ";
        if (tip != null)
            saida += "[" + tip + "]";
        else
            saida += "null";
        return saida;
    }
}
